### PR TITLE
CPDTP-232 Add a rake task for generating sandbox mentors

### PR DESCRIPTION
**Pull request by timabell**
### Context

In order for lead providers to be able to test sending declarations for mentors they will need some mentor ids to test again. This rake task is intended to be run against the sandbox environment to generate that test data.

The old sandbox seeds file was a one-time seed and is no longer needed so removed that too.

The mismatch of provider names in the sandbox will be handled separately in https://dfedigital.atlassian.net/browse/CPDTP-304

### Changes proposed in this pull request

* Add rake task for generating sandbox mentors
* Remove old sandbox seeds

#### Example output

```
tim@fox:~/repo/dfe/early-careers-framework(CPDTP-232-sandbox-mentor-seeds)                                                                                                                                         
$ bundle exec rails lead_provider:seed_mentors                                                                                                                                                                     
Generating seed mentors for 'Capita'...                                                                                                                                                                            
 Mentor with user_id 5a2916ce-ef6a-40bf-9ef1-426a502fc942 generated with 0 ECTs                                                                                                                                    
 Mentor with user_id 5e77d043-9f4b-444b-87c3-981724f4e6f1 generated with 1 ECTs                                                                                                                                    
 Mentor with user_id fec83784-ba4a-4b0d-bb1d-dfed33ebfd34 generated with 1 ECTs                                                                                                                                    
 Mentor with user_id 5b17cbce-a504-4f4a-988c-8f3e251d213b generated with 1 ECTs                                                                                                                                    
 Mentor with user_id c37d4b7e-522c-45ab-b162-65f06134964d generated with 1 ECTs                                                                                                                                    
 Mentor with user_id a941f61e-c65b-4d01-b53c-76d59399543f generated with 1 ECTs                                                                                                                                    
 Mentor with user_id 81bb2cd1-a808-412a-8b35-b5cbed46af58 generated with 1 ECTs                                                                                                                                    
 Mentor with user_id 509ee144-e567-42e3-91df-8361f1de1321 generated with 2 ECTs                                                                                                                                    
 Mentor with user_id 80857e21-bd74-43e0-bda2-53710493a91b generated with 2 ECTs                                                                                                                                    
 Mentor with user_id d192d413-a8fa-49a5-9515-e38f124b2212 generated with 2 ECTs                                                                                                                                    
 Before: 0 mentors, after: 10                                                                                                                                                                                      
 Before: 0 ECTs, after: 12                                                                                                                                                                                         
Generating seed mentors for 'Teach First'...                                                                                                                                                                       
 Mentor with user_id 76cad69c-f01a-437f-beb2-70dc814ab026 generated with 3 ECTs                                                                                                                                    
 Mentor with user_id cb5f1344-c669-4eac-9d74-54d625656c26 generated with 3 ECTs
 Mentor with user_id f86db2b9-8f9b-46c6-a9c7-a304c4f3525c generated with 1 ECTs
 Mentor with user_id 7a41be25-13a5-45e5-890b-98b411831ce2 generated with 0 ECTs
 Mentor with user_id 2077b08b-2d3c-46fe-af74-365dc8fdc6e8 generated with 1 ECTs
 Mentor with user_id e7fc5dd9-53c4-4ab4-9d2b-5a3582e38a04 generated with 2 ECTs
 Mentor with user_id d4f6b973-0eb3-4108-a59f-88a4ab1a6db0 generated with 1 ECTs
 Mentor with user_id 5859ae82-3c20-4dd8-bec3-201b4e7fa5e2 generated with 3 ECTs
 Mentor with user_id 4512da5d-e1d3-4d47-8d47-2fb026d523bb generated with 0 ECTs
 Mentor with user_id 1eaa9f62-0511-488b-b24e-8bd37876b59c generated with 0 ECTs
 Before: 0 mentors, after: 10
 Before: 0 ECTs, after: 14
...
```

### Guidance to review

```bash
bundle exec rails lead_provider:seed_mentors
```

### How can I view this in a review app?

Connect to the review app with `cf ssh app-name` and run the rake task. https://github.com/DFE-Digital/early-careers-framework/blob/develop/documentation/debugging_in_govpaas.md#rails-console

### CODE REVIEW

**Comment by ethanmills** [comment]
Would we be able to assign at least some of them some ECTs?

**Comment by timabell** [comment]
> Would we be able to assign at least some of them some ECTs?

We could but I'm not sure what it would give us for the task at hand.

File: lib/tasks/sandbox_data.rake [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    mentor = User.create!(full_name: Faker::Name.name, email: Faker::Internet.email)
+    MentorProfile.create!(user: mentor, school: school, cohort: cohort)
+    logger.info(" Mentor with user_id #{mentor.id} generated")
+  end
+  new_count = User.mentors_for_lead_provider(lead_provider).count
+  logger.info(" Before: #{existing_count}, after: #{new_count}")
+end
+
+def create_school_and_associations(lead_provider, cohort, index)
+  school = School.find_by!(
```
Comment (plisovin) [inline lib/tasks/sandbox_data.rake:35]: I think it's gonna fail if school doesn't exist and the previous sandbox seed file hasn't been run before

Comment (timabell) [reply lib/tasks/sandbox_data.rake:35]: It's only for the sandbox which it's been run on, do we care?

Comment (timabell) [reply lib/tasks/sandbox_data.rake:35]: Oh actually you can't run `db:reset` then this, okay I can see why we need it to find_or_create now. Fixing...
Thanks

Comment (kicferk1) [reply lib/tasks/sandbox_data.rake:35]: Is the fix `find_or_create_by!`? If so, then maybe we don't really care, but the fix is so small and simple that, well, why not?

Comment (timabell) [reply lib/tasks/sandbox_data.rake]: Done the schools etc, but this has flushed out a similar issue with the UCL provider, I need to look a bit closer at this....

**Comment by timabell** [comment]
> > Would we be able to assign at least some of them some ECTs?
> 
> We could but I'm not sure what it would give us for the task at hand.

Discussed further on slack...

Going to add random 0 to 3 ECTs per mentor to give realistic data shape that will then be visible to lead providers in the participants API.

File: lib/tasks/sandbox_data.rake [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+namespace :lead_provider do
+  desc "create seed mentors for API testing"
+  task seed_mentors: :environment do
+    logger = Logger.new($stdout)
+    logger.formatter = proc do |_severity, _datetime, _progname, msg|
+      "#{msg}\n"
+    end
+
+    providers_names = ["Capita", "Teach First", "UCL", "Best Practice Network", "Ambition", "Education Development Trust"]
```
Comment (timabell) [inline lib/tasks/sandbox_data.rake:11]: currently getting an error locally for UCL entry which needs investigating...

Comment (timabell) [reply lib/tasks/sandbox_data.rake]: also fixed Ambition

**Comment by timabell** [comment]
Provider names fixed, rebased on latest `develop`.

**Review by ethanmills — APPROVED** [review]

**Pull request merged**